### PR TITLE
[WFLY-15260] Support for securing the management console with OIDC

### PR DESCRIPTION
**Pull request by fjuma**
https://issues.redhat.com/browse/WFLY-15260

### CODE REVIEW

File: elytron/WFLY-15260_oidc_management_console.adoc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+to publish the OIDC configuration for the management console).
+** The previous `keycloak` subsystem published the OIDC configuration to be used for the
+management console via the http://localhost:MANAGEMENT_PORT/keycloak/adapter/wildfly-console/index.html
+endpoint. HAL would access this endpoint and if it found OIDC configuration, HAL would then
+redirect to the Keycloak login page. Similarly, the `elytron-oidc-client` subsystem will publish the
+OIDC configuration to be used for the management console to a new endpoint (e.g.,
+http://localhost:MANAGEMENT_PORT/elytron-oidc-client/wildfly-console/index.html). HAL will need to be
+updated to access this endpoint and if it finds OIDC configuration, HAL would then
+redirect to the OIDC provider login page.
+* It will be possible to log out of the management console after having logged in with OIDC.
```
Comment (OndrejKotek) [inline elytron/WFLY-15260_oidc_management_console.adoc:85]: Just local logout of the console, or the complete OIDC logout?

File: elytron/WFLY-15260_oidc_management_console.adoc [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+////
+
+== Security Considerations
+
+This is a security RFE.
+
+== Test Plan
+
+Tests will be added to the WildFly testsuite to verify that the management console
+can be secured successfully with OIDC.
```
Comment (OndrejKotek) [inline elytron/WFLY-15260_oidc_management_console.adoc:128]: Simple tests for the logout and for using RBAC would be also useful.

File: elytron/WFLY-15260_oidc_management_console.adoc (hunk 1 above)
Comment (fjuma) [reply elytron/WFLY-15260_oidc_management_console.adoc:85]: I think that with the existing support, logging out resulted in the complete OIDC logout. So the idea would be to do the same here.

File: elytron/WFLY-15260_oidc_management_console.adoc (hunk 2 above)
Comment (fjuma) [reply elytron/WFLY-15260_oidc_management_console.adoc:128]: Will update this section to mention that, thanks @OndrejKotek!

**Review by hpehl — APPROVED** [review]
Sorry @fjuma, for the delay. I missed that this still needs my review. 

The RFE looks good to me. No objections from my side.

File: elytron/WFLY-15260_oidc_management_console.adoc (hunk 1 above)
Comment (kstekovi) [reply elytron/WFLY-15260_oidc_management_console.adoc:85]: Will the logout have effect also on an application deployed into EAP secured by OIDC when we are logged in with the same user?

**Comment by kstekovi** [comment]
Looks good also to me.

**Pull request merged**